client.go

```diff
@@ -689,19 +692,26 @@ func newLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 		MaxMessageSize:     l.maxMessageSize,
 	}
 
+	// copy source and target
+	if l.source != nil {
+		attach.Source = l.source
+	}
+	if l.target != nil {
+		attach.Target = l.target
+	}
```

Comment:
The nil checks are unnecessary. Just assign to the `Source` and `Target` fields directly in the struct literal above.

Reply:
done